app/forms/update_user_password_form.rb

```diff
@@ -0,0 +1,26 @@
+class UpdateUserPasswordForm
+  include ActiveModel::Model
+  include FormPasswordValidator
+
+  attr_accessor :password
+  attr_reader :user
+
+  def persisted?
+    true
+  end
+
+  def initialize(user)
+    @user = user
+  end
+
+  def submit(params)
+    self.password = params[:password]
+
+    return false unless valid?
+    @user.update(params)
+  end
+
+  def mobile_changed?
+    false
+  end
```

Comment:
I see. That is not the ideal way to implement this, but Each form object and submission should only be concerned with itself. One of the reasons we replaced validations in the User model with Form Objects was to avoid running conditionals that will always return the same value. 

For example, when a user first signs up, they only need to enter their email, so we don't need to validate their phone number. We used to have a conditional that asked something like "Is the user already registered? If so, validate the phone number". That meant that we had to check if the user was registered every time the user was registering for the first time. The answer would always be "no", so why keep asking if we know for sure we don't need to validate the phone number at this step?

The same thing applies here. We know for sure that the mobile did not change if the user is only updating their email or their password. What I propose is to have a separate controller for each thing that needs to be updated, with `edit` and `update` methods to stick to Rails and REST conventions. So, instead of adding another method to the EditInfoController, I would create an `EditPasswordController` with `edit` and `update` methods. That way, the `get` and `put` requests are clearly separated.